model/event/event_finder.go

```diff
@@ -169,6 +159,13 @@ func DistroEventsForId(id string) db.Q {
 	return db.Query(filter)
 }
 
+func DistroAMIModifiedForId(id string) db.Q {
+	filter := ResourceTypeKeyIs(ResourceTypeDistro)
+	filter[ResourceIdKey] = id
+	filter[TypeKey] = EventDistroAMIModfied
+	return db.Query(filter).Sort([]string{"-" + TimestampKey}).Limit(1)
```

Comment:
Does this need an index? Or are distro events TTL'd?

Reply:
Good question; an index already exists for resource ID / resource type / timestamp but not including event type, which I believe would mean that the timestamp portion of the index wouldn't be used. I think a new index would be about 100 GB. That being said, there aren't very many distro events relatively, so I'm not sure it's worth an additional index when resource/resource type would narrow it down a lot. What do you think?

Reply:
Could we use a partial index to include just the events with distro AMI modified?

I don't think it's that big of a performance concern for now. If it becomes an issue, it would show up in the slow queries eventually.